<?xml version="1.0"?>
<launch>
<!-- <arg name="quad_names" default="hummingbird,quadrotor1,quadrotor2"/> --> 
    <arg name="num_quad" default= "5"/>
    <arg name="quad_name" default="hummingbird"/>
    <param name="mode" value="true" />


  <arg name="mav_name" default="$(arg quad_name)"/>
  <arg name="model" value="$(find rotors_description)/urdf/mav_generic_odometry_sensor.gazebo"/>
  <arg name="world_name" default="$(find rotors_gazebo)/worlds/basic.world"/>

  <arg name="x_init" default="0.0"/>
  <arg name="y_init" default="0.0"/>

  <arg name="enable_logging" default="false" />
  <arg name="enable_ground_truth" default="true" />
  <arg name="enable_command_feedthrough" default="false"/>
  <arg name="log_file" default="$(arg mav_name)" />
  <arg name="paused" value="true"/>
  <arg name="gui" value="true"/>
  <arg name="use_ground_truth" value="true"/>
  <arg name="verbose" default="false"/>
  <arg name="debug" default="false"/>

  <!-- Basic simulation environment !-->
  <include file="$(find cnuav_control)/launch/basics/base_quad_simulation.launch">
    <arg name="quad_name" value="$(arg quad_name)"/>
    <arg name="world_name" value="$(arg world_name)"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="enable_command_feedthrough" value="$(arg enable_command_feedthrough)" />

    <arg name="mav_name" value="$(arg mav_name)"/>
    <arg name="model" value="$(arg model)"/>
    <arg name="enable_logging" value="$(arg enable_logging)"/>
    <arg name="enable_ground_truth" value="$(arg enable_ground_truth)"/>
    <arg name="log_file" value="$(arg log_file)"/>
    <arg name="verbose" default="$(arg verbose)"/>
    <arg name="debug" default="$(arg debug)"/>

    <arg name="x_init" value="$(arg x_init)"/>
    <arg name="y_init" value="$(arg y_init)"/>
  </include>


      <!-- 启动 odometry_visualizer 节点 -->
    <!-- <node name="odometry_visualizer" pkg="cnuav_control" type="odometry_visualizer" output="screen" /> -->

  <!-- 启动 RViz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find cnuav_control)/rviz/blanck.rviz" output="screen" />

    <!-- 发布 world frame -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="world_to_odom_broadcaster" args="0 0 0 0 0 0 1 world odom" />



    <node pkg="cnuav_control" type="odometry_visualizer" name="odometry_visualizer" output="screen">
      <remap from="pose" to="/$(arg quad_name)/ground_truth/odometry" />
      <remap from="marker" to="/$(arg quad_name)/pose/marker"/>
      <!-- <remap from="traj" to="/$(arg quad_name)/trajectory"/> -->
      <remap from="traj" to="/$(arg quad_name)/autopilot/trajectory"/>
      <remap from="traj_marker" to="/$(arg quad_name)/traj_marker"/>
      <param name="mode" value="true" />
      <param name="quad_name" value="$(arg quad_name)" />
    </node>


</launch>
